### PR TITLE
Replace ThreadPoolExecutor creation with ThreadUtils API [databricks]

### LINKED ISSUE #10759: Use Spark ThreadUtils to create thread pools
There are a number of places in the code where thread pool executors are created manually.  These should be switched to leverage Spark's ThreadUtils class, as some Spark platforms may have custom logic that needs to be leveraged when new threads are created.

### PR DESCRIPTION
**Pull request by razajafri**
This PR replaces the manual thread pool creation with the `TrampolieUtil` API which uses the `org.apache.spark.util.ThreadUtils` under the hood. 

fixes #10759

### CODE REVIEW

File: shuffle-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/ucx/UCXShuffleTransport.scala [hunk 1]
```diff
@@ -240,21 +241,27 @@ class UCXShuffleTransport(shuffleServerId: BlockManagerId, rapidsConf: RapidsCon
...
+      rapidsConf.shuffleClientThreadKeepAliveTime)
+    threadPoolExecutor.setCorePoolSize(1)
+    threadPoolExecutor.setThreadFactory(
+      GpuDeviceManager.wrapThreadFactory(
+        new ThreadFactoryBuilder()
+          .setNameFormat("shuffle-transport-client-exec-%d")
+          .setDaemon(true)
+          .build,
+        null,
+        () => RmmSpark.removeAllCurrentThreadAssociation()))
```
Comment (jlowe) [inline shuffle-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/ucx/UCXShuffleTransport.scala:259]: Ignoring the existing thread factory will likely defeat the point of using ThreadUtils to create the thread pool.  Any per-thread logic provided by the Spark platform is highly likely placed in the thread factory that ThreadUtils is setting up.
```suggestion
    threadPoolExecutor.setThreadFactory(
      GpuDeviceManager.wrapThreadFactory(
        threadPoolExecutor.getThreadFactory,
        null,
        () => RmmSpark.removeAllCurrentThreadAssociation()))
```

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastExchangeExec.scala [hunk 2]
```diff
@@ -514,33 +514,8 @@ abstract class GpuBroadcastExchangeExecBase(
...
-      TimeUnit.SECONDS,
-      new LinkedBlockingQueue[Runnable],
-      threadFactory)
-    threadPool.allowCoreThreadTimeOut(true)
-    threadPool
-  }
-
   val executionContext = ExecutionContext.fromExecutorService(
-    newDaemonCachedThreadPool("gpu-broadcast-exchange",
+    TrampolineUtil.newDaemonCachedThreadPool("gpu-broadcast-exchange",
```
Comment (jlowe) [inline sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastExchangeExec.scala:518]: Don't need TrampolineUtil, this is already in a package under org.apache.spark.
```suggestion
    ThreadUtils.newDaemonCachedThreadPool("gpu-broadcast-exchange",
```

**Comment by razajafri** [comment]
@jlowe I have addressed your concern. PTAL

**Review by jlowe — APPROVED** [review]

**Comment by jlowe** [comment]
build

**Pull request merged**